pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package upgradeconverter
+
+import (
+	"flag"
+	log "github.com/sirupsen/logrus"
+)
+
+// ConverterAction - Defines the types of action that can occur when moving between versions
+type ConverterAction uint8
+
+const (
+	// Convert - Convert from old to new
+	Convert ConverterAction = iota + 1
+	// CleanupOld - Delete old versions
+	CleanupOld
+	// CleanupNew - Delete new versions if outdated
+	CleanupNew
+	// NoAction - Do nothing
+	NoAction
+)
+
+var conversionHandlers = []ConversionHandler{
+	{
+		description: "Convert Global Settings to new format",
+		handlerFunc: handleUpgradeGlobalConfig,
+	},
+}
+
+// Run - runs the main upgradeconverter process
+func Run() {
+	actionStr := flag.String("actionStr", "", "Choose an action from convert/cleanupOld/cleanupNew")
```

Comment:
Why such a long name for the argument?
A valid call would be
./upgradeconverter -actionStr=convert -oldVersionPath=/persist/config/.. -newVersionPath=X

I think this can be ./upgradeconverter -a convert -o /persist/config/.. -n X
IMNSHO and camelCase isn't for users; its for programmers.